core/src/main/java/feign/FeignException.java

```diff
@@ -61,10 +64,12 @@ protected FeignException(int status, String message) {
     this.request = null;
   }
 
-  protected FeignException(int status, String message, byte[] responseBody) {
+  protected FeignException(int status, String message, byte[] responseBody,
+      Map<String, Collection<String>> responseHeaders) {
     super(message);
     this.status = status;
     this.responseBody = responseBody;
+    this.responseHeaders = caseInsensitiveCopyOf(responseHeaders);
```

Comment:
I don't think `FeignException` should re-order response headers.  It may make sense to sort it, sure, but not here.